code-challenges.test.js

```diff
@@ -15,24 +17,74 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+describe("removeShuffle", () => {
+    const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    const expected1 = ["yellow", "blue", "pink", "green"]
+    ////////////////////////////////////////////////////////////////////////////////////////
+    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    const expected2 = ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+    it('removes the first item from the array and shuffles the remaining items', () => {
+        expect(removeShuffle(colors1)).toEqual(expect.arrayContaining(expected1))
+        expect(removeShuffle(colors2)).toEqual(expect.arrayContaining(expected2))
+    })
+})
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// input: array 
+// output: return an array with the first element removed as well as shuffling the updated array
+const removeShuffle = (array) => { // create a functions to pass an array
+    //use.shift() to remove the first element from the array
+    array.shift() 
+    // create an empty array to store numbers
+    let arrayPos = [] 
+    // create an empty array to store the new order of the original array
+    let newArr = []
+
+    // create a while loop to store numbers to arrayPos
+    while (arrayPos.length < array.length){
+        // generate a new number from 0 to the length of the array
+        let randomNum = Math.floor((Math.random() * array.length) )
+        // if the number is not in arrayPos
+        if (!arrayPos.includes(randomNum)){
+            //then store that number into arrayPos
+            arrayPos.push(randomNum)
+        }
+    }
+    //create a loop to change the position of the elements from the original array
+    for (let i=0; i < arrayPos.length; i++){
+        //push the element into the new array
+      newArr.push(array[arrayPos[i]])
+    }
+    // return newArr to which it was called
+    return(newArr)
+  }
```

Comment:
Way to really think through every step.  Very impressive, even though I watched you do it. :) 🎉